File: frontend/src/types/sale.ts
export interface SaleEvent {
  id: string;
  category: "flight" | "hotel" | "package";
  title: string;
  description?: string;
  sale_start?: string;
  sale_end?: string;
  travel_start?: string;
  travel_end?: string;
  discount_rate?: number;
  min_price_jpy?: number;
  target_routes?: { origin: string; destination: string }[];
  target_destinations?: string[];
  coupon_code?: string;
  booking_url?: string;
  source: string;
  is_verified: boolean;
  created_at: string;
}

export interface SaleListResponse {
  total: number;
  items: SaleEvent[];
}

File: backend/schemas/sale.py
from pydantic import BaseModel
from typing import Optional, List, Any
from datetime import date, datetime
import uuid


class SaleEventOut(BaseModel):
    id: uuid.UUID
    category: str
    title: str
    description: Optional[str] = None
    sale_start: Optional[datetime] = None
    sale_end: Optional[datetime] = None
    travel_start: Optional[date] = None
    travel_end: Optional[date] = None
    discount_rate: Optional[float] = None
    min_price_jpy: Optional[int] = None
    target_routes: Optional[Any] = None
    target_destinations: Optional[Any] = None
    coupon_code: Optional[str] = None
    booking_url: Optional[str] = None
    source: str
    is_verified: bool
    created_at: datetime

    class Config:
        from_attributes = True


class SaleListResponse(BaseModel):
    total: int
    items: List[SaleEventOut]
